src/TorchSharp/Utils/TensorAccessor.cs

```diff
@@ -44,11 +44,119 @@ public TensorAccessor(torch.Tensor tensor)
 
         public T[] ToArray()
         {
+            if (_tensor.ndim < 2)
+                return (T[])ToNDArray();
+
             var result = new T[Count];
             CopyTo(result);
             return result;
         }
 
+        /// <summary>
+        /// Extract tensor data as a multi-dimensional .NET array, with the same number of dimensions as the tensor.
+        /// </summary>
+        /// <returns>An array object, which should be cast to the concrete array type.</returns>
+        /// <exception cref="NotImplementedException">Thrown if the tensor has more than 6 dimensions.</exception>
+        public System.Array ToNDArray()
```

Comment:
The nested loops in this method are necessary in order to apply the right strides. A tensor may be a view of another, and it is not correct to assume that storage is contiguous.

Reply:
So, is there something wrong with my PR?

Reply:
So, is there something wrong with my PR?

Reply:
I added this comment for reviewers. Before accepting your PR, I would like to see the relative performance -- the use reflection `array.SetValue()` and the repeated calls to 'TranslateIndex' are likely to have some impact on performance, possibly significant. 

As @xhuan8 pointed out, a simple index++ should work, except that the strides need to be taken in to account. Even if the reflection-based approach is slow, it could potentially be used to cover additional, less common dimensionality. Rather than calling TranslateIndex(), a solution based on recursion could be used (SetValue() would still have to be used).

Reply:
And on a procedural note -- your PR will have to be against this repo, not mine, since any submission will have to be subject to the TorchSharp CLA, which would be missed if I take your contribution into my personal repo.

Reply:
OK

Reply:
I've merged this PR now -- if you want to generalize the ToNDArray() to N dimensions, please submit a PR against this repo, so that we can handle the CLA properly.

Thanks for your interest and help!